src/relay/pass/enable_tensorrt.cc

```diff
@@ -0,0 +1,585 @@
+        }
+      }
+    }
+    if (op_name == "sum" || op_name == "prod" || op_name == "max" ||
+        op_name == "min" || op_name == "mean") {
+      if (const auto* attrs = call->attrs.as<ReduceAttrs>()) {
+        if (!attrs->axis.defined() || attrs->axis.size() == 0) {
+          compatible_ = false;
+          LOG(INFO) << op_name << " not supported: cannot reduce to scalar.";
+        } else {
+          for (size_t i = 0; i < attrs->axis.size(); i++) {
+            if (attrs->axis[i].as<IntImm>()->value == 0) {
+              compatible_ = false;
+              LOG(INFO) << op_name
+                        << " not supported: can't modify batch dimension.";
+              break;
+            }
+          }
+        }
+        if (attrs->exclude) {
+          compatible_ = false;
+          LOG(INFO) << op_name << " not supported: exclude not supported.";
+        }
+      }
+    }
+    if (op_name == "strided_slice") {
+      auto shape = GetShape(call->type_args[0]);
+      if (const auto* attrs = call->attrs.as<StridedSliceAttrs>()) {
+        if (attrs->begin[0].as<IntImm>()->value != 0 ||
+            (attrs->end[0].as<IntImm>()->value != -1 &&
+             attrs->end[0].as<IntImm>()->value != shape[0])) {
+          compatible_ = false;
+          LOG(INFO) << op_name
+                    << " not supported: can't modify batch dimension.";
+        }
+        for (size_t i = 0; i < attrs->begin.size(); i++) {
+          if (attrs->begin[i].as<IntImm>()->value < 0 ||
+              attrs->end[i].as<IntImm>()->value < 0) {
+            compatible_ = false;
+            LOG(INFO) << op_name
+                      << " not supported: start/end values must be positive.";
+          }
+        }
+      }
+    }
+    if (op_name == "adaptive_max_pool2d" || op_name == "adaptive_avg_pool2d") {
+      if (const auto* attrs = call->attrs.as<AdaptivePool2DAttrs>()) {
+        if ((attrs->output_size.size() == 1 &&
+             attrs->output_size[0].as<IntImm>()->value != 1) ||
+            (attrs->output_size.size() == 2 &&
+             (attrs->output_size[0].as<IntImm>()->value != 1 ||
+              attrs->output_size[1].as<IntImm>()->value != 1))) {
+          compatible_ = false;
+          LOG(INFO) << op_name << " not supported: output size must be (1, 1).";
+        }
+      }
+    }
+    if (op_name == "resize") {
+      if (const auto* attrs = call->attrs.as<ResizeAttrs>()) {
+        if (attrs->method != "nearest_neighbor" && attrs->method != "bilinear") {
+          compatible_ = false;
+          LOG(INFO) << op_name << " not supported: method must be nearest_neighor or bilinear";
+        }
+      }
+    }
+  }
+
+  bool Check(const Expr& expr) {
+    compatible_ = true;
+    VisitExpr(expr);
+    return compatible_;
+  }
+
+ private:
+  bool compatible_;
+  std::unordered_set<std::string> trt_compatible_ops;
+  std::tuple<int, int, int> trt_version_;
+};
+
```

Comment:
probably add a TODO here to indicate that this enabler can be removed or unified with GraphPartition after it is merged in the upstream.